docs/apache-airflow/best-practices.rst

```diff
@@ -54,6 +59,12 @@ Some of the ways you can avoid producing a different result -
     You should define repetitive parameters such as ``connection_id`` or S3 paths in ``default_args`` rather than declaring them for each task.
     The ``default_args`` help to avoid mistakes such as typographical errors.
 
+Creating a custom Operator
+---------------------------
+
+When implementing custom operators, do not make any expensive expensive operations in their ``__init__``. They are
+going to be instantiated once per scheduler run per task using them, and making database calls can significantly slow
+down scheduling and waste resources.
```

Comment:
This is probably something we need to mention in https://github.com/apache/airflow/blob/master/docs/apache-airflow/howto/custom-operator.rst#creating-a-custom-operator and create a ref link there.

Reply:
Got it!